Modelica/Blocks/Routing.mo

```diff
@@ -237,6 +237,12 @@ equation
 
   y = if not allowOutOfRange or index > 0 and index <= nin then
               k*u else outOfRangeValue;
+
+  if not allowOutOfRange then
+    assert(index <= nin and index > 0,
+          "Extractor: Selected input is out of range. Please check input \"index\"");
```

Comment:
Why not include the current value of index (and nin)? You cannot really check the input since after assertion the values are lost to the void.